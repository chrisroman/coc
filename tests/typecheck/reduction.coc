let idT = [a:*][b:a]a in
let id = (\c:*)(\d:c)d in
let fn = (\e: idT)e in
(fn id)
