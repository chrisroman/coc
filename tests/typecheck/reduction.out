let idT = [a: *] [b: a] a in 
let id = (λc: *) (λd: c) d in 
let fn = (λe: idT) e in 
(fn id)
{idT} typecheck_single_term [a: *] [b: a] a
Type: *
{id} typecheck_single_term (λc: *) (λd: c) d
Type: [c: *] [d: c] c
{fn} typecheck_single_term (λe: [a: *] [b: a] a) e
Type: [e: [a: *] [b: a] a] [a: *] [b: a] a
typecheck_single_term ((λe: [a: *] [b: a] a) e (λc: *) (λd: c) d)
Type: [a: *] [b: a] a
Value: (λc: *) (λd: c) d
